pkg/security/grantvmgroupaccess.go

```diff
@@ -0,0 +1,285 @@
+	inheritModeSubContainersOnlyInherit       inheritMode = 0x2
+	inheritModeSubContainersAndObjectsInherit inheritMode = 0x3
+	inheritModeNoPropagate                    inheritMode = 0x4
+	inheritModeInheritOnly                    inheritMode = 0x8
+	inheritModeInheritedAce                   inheritMode = 0x10
+
+	objectTypeFileObject objectType = 0x1
+
+	securityInformationOwner securityInformation = 0x1
+	securityInformationGroup securityInformation = 0x2
+	securityInformationDACL  securityInformation = 0x4
+	securityInformationSacl  securityInformation = 0x8
+	securityInformationLabel securityInformation = 0x10
+
+	shareModeRead   shareMode = 0x1
+	shareModeWrite  shareMode = 0x2
+	shareModeDelete shareMode = 0x4
+
+	sidVmGroup                   = "S-1-5-83-1-3166535780-1122986932-343720105-43916321"
+	sidVmWorkerProcessCapability = "S-1-15-3-1024-2268835264-3721307629-241982045-173645152-1490879176-104643441-2915960892-1612460704"
+
+	trusteeFormIsSid            trusteeForm = 0
+	trusteeFormIsName           trusteeForm = 1
+	trusteeFormBadForm          trusteeForm = 2
+	trusteeFormIsObjectsAndSid  trusteeForm = 3
+	trusteeFormIsObjectsAndName trusteeForm = 4
+
+	trusteeTypeUnknown        trusteeType = 0
+	trusteeTypeUser           trusteeType = 1
+	trusteeTypeGroup          trusteeType = 2
+	trusteeTypeDomain         trusteeType = 3
+	trusteeTypeAlias          trusteeType = 4
+	trusteeTypeWellKnownGroup trusteeType = 5
+	trusteeTypeDeleted        trusteeType = 6
+	trusteeTypeInvalid        trusteeType = 7
+	trusteeTypeComputer       trusteeType = 8
+)
+
+// GrantVMGroupAccess sets the DACL for the group SIDs to the specified
+// file or directory. This is a golang re-implementation of the same function
+// in vmcompute, just not exported in RS5. Which kind of sucks. Sucks a lot :/
+func GrantVmGroupAccess(name string) error {
+
+	// Stat (we use this to determine if file/directory is a directory).
+	s, err := os.Stat(name)
+	if err != nil {
+		return errors.Wrapf(err, "%s os.Stat %s", gvmga, name)
+	}
+
+	// Get a handle to the file/directory.
+	fd, err := createFile(name, s.IsDir())
+	if err != nil {
+		return err // already wrapped
+	}
+
+	// Make sure the handle is closed at the end.
+	defer syscall.CloseHandle(fd)
+
+	// Get the current DACL.
+	ot := objectTypeFileObject
+	si := securityInformationDACL
+	origDACL := uintptr(0)
+	if err := GetSecurityInfo(fd, uint32(ot), uint32(si), nil, nil, &origDACL, nil, nil); err != nil {
+		return errors.Wrapf(err, "%s GetSecurityInfo %s", gvmga, name)
+	}
+
+	// I don't understand why if retrieving the DACL, there is need to call
+	// LocalFree, but there is if retrieving the SD (which contains the DACL
+	// as part of it). Although the HCS implementation of GrantVmGroupAccess
+	// manipulates the SD, it's MUCH easier in go to manipulate the DACL
+	// to configure the relevant ACEs.
+	// defer syscall.LocalFree((syscall.Handle)(unsafe.Pointer(origDACL)))
```

Comment:
Dont you just want to pass `origDACL`. Not a pointer to it? It is already holding the value of the `HLOCAL`

Reply:
Confirmed by @jstarks that this isn't needed and won't leak. Have just removed this comment.